app-creator/main.go

```diff
@@ -0,0 +1,119 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+	"modules/handlers/templates"
+	"regexp"
+	"github.com/rs/cors"
+	"cloud.google.com/go/datastore"
+)
+
+func main() {
+	/**
+	* Getting application context.
+	*/
+	ctx := context.Background()
+
+	/** 
+	* Creating a datastore client. This instance is shared across the application.
+	*/
+	db, err := datastore.NewClient(ctx, os.Getenv("GOOGLE_CLOUD_PROJECT"))
+	if err != nil {
+		log.Fatal("Database connection error", err)
+	}
+
+	/**
+	* Creating shared logger instance.
+	*/
+	l := log.New(os.Stdout, "app-creator", log.LstdFlags)
+
+	/**
+	* Check if a PORT env variable is set.
+	*/
+	port := os.Getenv("PORT")
+
+	/**
+	* Strip any leading colons and make sure it only contains numbers.
+	*/
+	re := regexp.MustCompile(`^[:]([0-9]+)$`)
+	
+	/**
+	* Default to port 8080 if environment variable is not set or is invalid.
+	*/
+	if port == "" || !re.MatchString(port) {
+		port = ":8080"
+		log.Printf("Defaulting to port %s", port)
+	}
+	
+	/**
+	* Initializing a new server mux to handle routes.
+	*/
+	serverMux := http.NewServeMux()
+	
+	/**
+	* Serving static files.
+	*/
+	fs := http.FileServer(http.Dir("./static"))
+	serverMux.Handle("/", fs)
+	serverMux.Handle("/static/", http.StripPrefix("/static/", http.FileServer(http.Dir("./static"))))
+	
+	/**
+	* Setting up handlers.
+	*/
+	templatesHandler := handlers.NewTemplatesHandler(l, db)
+
+	/**
+	* Registering handlers.
+	*/
+	serverMux.Handle("/api/v1/templates", templatesHandler)
+
+	/**
+	* CORS middleware. Default option allows all origins. 
```

Comment:
Add an environment variable that, if present, only allows CORS for those domains (in a comma separated list). That way, in the App Engine yaml, we can add the appspot URL and limit the  XSS attack vector.